lib/ansible/modules/network/avi/avi_cloudconnectoruser.py

```diff
@@ -0,0 +1,121 @@
+# module_check: supported
+# Avi Version: 17.1.1
+#
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: avi_cloudconnectoruser
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: Module for setup of CloudConnectorUser Avi RESTful Object
+description:
+    - This module is used to configure CloudConnectorUser object
+    - more examples at U(https://github.com/avinetworks/devops)
+requirements: [ avisdk ]
+version_added: "2.4"
+options:
+    state:
+        description:
+            - The state that should be applied on the entity.
+        default: present
+        choices: ["absent","present"]
+    name:
+        description:
+            - Name of the object.
+        required: true
+    private_key:
+        description:
+            - Private_key of cloudconnectoruser.
+    public_key:
+        description:
+            - Public_key of cloudconnectoruser.
+    tenant_ref:
+        description:
+            - It is a reference to an object of type tenant.
+    url:
+        description:
+            - Avi controller URL of the object.
+    uuid:
+        description:
+            - Unique object identifier of the object.
+extends_documentation_fragment:
+    - avi
+'''
+
+
+EXAMPLES = '''
+  - name: Create a Cloud connector user that is used for integration into cloud platforms
+    avi_cloudconnectoruser:
+      controller: ''
+      name: root
+      password: ''
+      private_key: |
+        -----BEGIN RSA PRIVATE KEY-----
+        -----END RSA PRIVATE KEY-----'
+      public_key: 'ssh-rsa ...'
+      tenant_ref: admin
+      username: ''
+'''
+RETURN = '''
+obj:
+    description: CloudConnectorUser (api/cloudconnectoruser) object
+    returned: success, changed
+    type: dict
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from ansible.module_utils.avi import (
+        avi_common_argument_spec, HAS_AVI, avi_ansible_api)
+except ImportError:
+    HAS_AVI = False
+
+
+def main():
+    argument_specs = dict(
+        state=dict(default='present',
+                   choices=['absent', 'present']),
+        name=dict(type='str', required=True),
+        private_key=dict(type='str',),
```

Comment:
I wonder if this (and any other instance of `private_key`) should have `no_log=true`

I might have missed this in some other PRs, so feel free to raise one PR that updates all the modules.